### PR TITLE
refactor: introduce details-view-action-creator

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Refactor `ActionCreator` by extracting functions related to `DetailsViewActions` (no other actions involved) to a new `DetailsViewActionCreator`.

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
   - does not apply
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
   - does not apply
- [x] (UI changes only) Verified usability with NVDA/JAWS
   - does not apply

### CODE REVIEW

File: src/tests/unit/tests/background/actions/details-view-action-creator.test.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        detailsViewControllerMock.verify(controller => controller.showDetailsView(tabId), Times.once());
+        telemetryEventHandlerMock.verify(handler => handler.publishTelemetry(SETTINGS_PANEL_OPEN, payload), Times.once());
+    });
+
+    test('on SettingsPanel.ClosePanel', () => {
+        const closeSeetingsPanelMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['closeSettingsPanel']).returns(() => closeSeetingsPanelMock.object);
+
+        const payload: BaseActionPayload = {
```
Comment (waabid) [inline src/tests/unit/tests/background/actions/details-view-action-creator.test.ts]: Seems to be the same payload as above; maybe you can move this to beforeEach.

File: src/tests/unit/tests/background/actions/details-view-action-creator.test.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        setSelectedPanelMock.verify(action => action.invoke(payload), Times.once());
+    });
+
+    test('on Visualization.DetailsView.GetState', () => {
+        const getCurrentStateMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['getCurrentState']).returns(() => getCurrentStateMock.object);
+
+        registerCallbackMock
```
Comment (waabid) [inline src/tests/unit/tests/background/actions/details-view-action-creator.test.ts]: Might not be a bad idea to move this to a private method; seems duplicated in this test file.

Comment (smoralesd) [reply src/tests/unit/tests/background/actions/details-view-action-creator.test.ts]: the setting of the `registerCallbackMock`?? we have 3 different ways of calling the listener (no params, 1 and 2 params). Tests are short enough that I don't think this is a big deal. I'm planning on coming back to the action creator tests (once I finish breaking up the current action creator and global action creator) and apply something similar to what we have for the original action creator, i.e. an action creator tester class. But that seems to be out of scope for this PR.

File: src/tests/unit/tests/background/actions/details-view-action-creator.test.ts (hunk 1 above)
Comment (smoralesd) [reply src/tests/unit/tests/background/actions/details-view-action-creator.test.ts]: Updated.

**Pull request merged**